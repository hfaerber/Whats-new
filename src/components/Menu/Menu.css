.menu {
  grid-area: menu;
  color: whitesmoke;
  /* width: 20%; */
  display: flex;
  flex-direction: column;
  padding: 10px;
}

.navButton {
  padding: 20px;
  width: 100%;
  font-size: 1.2em;
  background-color: #141527;
  color: whitesmoke;
  border: none;
  margin: 0;
  cursor: pointer;
}
