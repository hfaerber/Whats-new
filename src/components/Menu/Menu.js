import React from 'react';
import './Menu.css'
import PropTypes from 'prop-types';

const Menu = ({setSelectedNews}) => {
  return(
    <div className='menu'>
      <nav onClick={(event) =>
        setSelectedNews(event.target.name)}>
        <button name='local' className='navButton'>
          Local News</button>
        <button name='technology' className='navButton'>
          Technology</button>
        <button name='entertainment' className='navButton'>
          Entertainment</button>
        <button name='science' className='navButton'>
          Science</button>
        <button  name='health' className='navButton'>
          Health</button>
      </nav>
    </div>
  )
}

Menu.propTypes = {
  setSelectedNews: PropTypes.func,
}

export default Menu;
