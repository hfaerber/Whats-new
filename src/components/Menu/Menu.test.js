import React from 'react';
import Menu from './Menu';
import { shallow } from 'enzyme';

describe('Menu', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<Menu
      setSelectedNews={jest.fn()}
    />);
    expect(wrapper).toMatchSnapshot();
  });
})
