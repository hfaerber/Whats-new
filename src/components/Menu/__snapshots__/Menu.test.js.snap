// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot with all data passed in correctly 1`] = `
<div
  className="menu"
>
  <nav
    onClick={[Function]}
  >
    <button
      className="navButton"
      name="local"
    >
      Local News
    </button>
    <button
      className="navButton"
      name="technology"
    >
      Technology
    </button>
    <button
      className="navButton"
      name="entertainment"
    >
      Entertainment
    </button>
    <button
      className="navButton"
      name="science"
    >
      Science
    </button>
    <button
      className="navButton"
      name="health"
    >
      Health
    </button>
  </nav>
</div>
`;
